code/game/objects/items.dm

```diff
@@ -676,6 +680,15 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 /obj/item/proc/on_mob_death(mob/living/L, gibbed)
 
+/obj/item/proc/grind_requirements(obj/machinery/reagentgrinder/R) //Used to check for extra requirements for grinding an object
+	return TRUE
+
+ //Called BEFORE the object is ground up - use this to change grind results based on conditions
+ //Use "return -1" to prevent the grinding from occurring
+/obj/item/proc/on_grind()

```

Comment:
These functions should, by default, return `grind_results` and `juice_results` or a -1 if the list is null or empty. 

That way, subtypes can override these functions for more complicated stuff, like returning the reagents list but modified

Reply:
`on_grind()` exists in order *to* change the reagents list somehow before grinding the object. What you described seems to be the already-existing function of the proc.